### PR TITLE
Refactor the sequential-challenge element and app-element

### PR DESCRIPTION
**Pull request by victorbalan**

### CODE REVIEW

File: app/elements/elements.html [hunk 1]
```diff
@@ -22,3 +22,5 @@
 
 <!-- TODO(victorbalan): investigate why this element cause the flickering -->
 <link rel="import" href="/elements/skill/skill-list.html">
+
+<link rel="import" href="/elements/app-element.html">
```
Comment (pbochis) [inline app/elements/elements.html]: why is this needed? You import `elements.html` [here](https://github.com/coduno/app/pull/112/files#diff-b89d9c11e31f0460fcd494ea9ec9764eR1) and in the `index.html` you import `app-element.html`. This looks like a useless import, and it also creates a circular dependency(which i don't know if it can cause any problems but better safe than sorry)

File: app/elements/app-element.html [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+					type: Object,
+					notify: true
+				},
+				pendingRequests: {
+					type: Number,
+					value: 0
+				}
+			},
+			ready: function(){
+				this.addEventListener('dom-change', function() {
```
Comment (Exordian) [inline app/elements/app-element.html]: ?

File: app/elements/elements.html (hunk 1 above)
Comment (Exordian) [inline app/elements/elements.html]: since the app element should be the root elements, it will be (in my opinion) always available and thus must not be in any "import common" files

File: app/elements/services/cookie-service.html [hunk 3]
```diff
@@ -30,6 +30,9 @@
 					observer: '_userChanged'
 				}
 			},
+			ready: function(){
+				this.app = document.querySelector('#app');
```
Comment (Exordian) [inline app/elements/services/cookie-service.html]: `this.app` isn't used in here?

File: app/elements/services/profile-service.html [hunk 4]
```diff
@@ -28,16 +28,19 @@
 					reflectToAttribute: true
 				}
 			},
+			ready: function(){
+				this.app = document.querySelector('#app');
```
Comment (Exordian) [inline app/elements/services/profile-service.html]: is this really needed in every element/service?

File: app/elements/app-element.html [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+							<template is="dom-if" if="{{isCoder}}">
+								<paper-tab link><a href="/challenge" class="center-center horizontal layout fullHeight">Challenge</a></paper-tab>
+							</template>
+							<span class="flex-5"></span>
+						</paper-tabs>
+					</div>
+					<paper-menu-button horizontal-align="right">
+						<paper-icon-button icon="supervisor-account" class="dropdown-trigger"></paper-icon-button>
+						<paper-menu class="dropdown-content">
+								<paper-item><a href="/" style="color: #444444; text-decoration: none;">Settings</a></paper-item>
```
Comment (lorenzleutgeb) [inline app/elements/app-element.html]: Wrong indent

File: app/elements/submission/challenge-nav.html [hunk 6]
```diff
@@ -62,10 +62,18 @@
 					notify: true,
 					reflectToAttribute: true
 				},
+				taskOffset: {
```
Comment (Exordian) [inline app/elements/submission/challenge-nav.html]: is this `taskTimeLeft` ? `taskOffset` seems unintuitive

File: app/elements/submission/sequential/sequential-challenge.html [hunk 7]
```diff
@@ -105,6 +85,14 @@
 				taskDuration: {
 					type: Number,
 					notify: true
+				},
+				challengeOffset: {
+					type: Number,
+					value: 0
+				},
+				taskOffset: {
```
Comment (Exordian) [inline app/elements/submission/sequential/sequential-challenge.html]: same as above

File: app/elements/submission/tasks/general-code-task.html [hunk 8]
```diff
@@ -164,10 +164,6 @@
 				this.control.loading();
 			},
 			finishTask: function(){
-				var msg = 'Are you sure to finish this task? You will not able to change come back again.';
-				if (!window.confirm(msg)) {
```
Comment (Exordian) [inline app/elements/submission/tasks/general-code-task.html:168]: Do not just delete random code :(

File: app/elements/services/profile-service.html (hunk 4 above)
Comment (victorbalan) [inline app/elements/services/profile-service.html]: yes, with it we start the progress bar and currently polymer does not allow us to implement custom elements..

File: app/elements/submission/tasks/general-code-task.html (hunk 8 above)
Comment (victorbalan) [inline app/elements/submission/tasks/general-code-task.html:168]: i did not remove it. I moved it [here](https://github.com/coduno/app/pull/112/files#diff-198e20b282f93443c6266aa1a67f21ffR147)

**Comment by pbochis** [comment]
rebased and pushed in #126
